### ISSUE TITLE
Application crash when moving/changing state on a Window after calling Close on WebView2 before it's unloaded

### ISSUE THREAD

**Issue by NicholasChrzan** [microsoft/microsoft-ui-xaml#10219]
### Describe the bug

If you call Close on a WebView2 control before the control has been unloaded and change the position of the window (changing Overlapped presenter in our case) the app will crash.

The root cause of this looks like the AppWindow.Changed event is unwired in the Unloaded event and not in the Close method. When the window position changes the NotifyParentWindowPositionChanged is called on the CoreWebViewController which causes an access violation which crashes the application.

A workaround is to call the Close in the Unloaded event of the WebView2.

### Steps to reproduce the bug

1. Maximize window (OverlappedPresenter.Maximize)
2. Call Close on WebView
3. Restore Window (OverlappedPresenter.Restore)

### NuGet package version

WinUI 3 - Windows App SDK 1.6.3: 1.6.241114003

### Windows version

Windows 11 (23H2): Build 22631

**Comment by karkarl** [comment]
This seems to be a dupe of #8395 that should have been resolved here: https://github.com/microsoft/microsoft-ui-xaml/issues/8395#issuecomment-1730050026

Can you provide a crash dump or repro app, this may be a regression.

**Comment by NicholasChrzan** [comment]
@karkarl We found this crash when moving from 1.4->1.6. I do have a crash dump how can I upload to you (it's not letting me attach it here).

**Comment by NicholasChrzan** [comment]
Also. to clarify this isn't related to Window.Close() it's related to the WebView2.Close() method.

For looking at this in WinDbg:

The crash happens here when calling m_coreWebViewController.NotifyParentWindowPositionChanged();

![Image](https://github.com/user-attachments/assets/06cb6787-a640-4f6a-9115-a45dbebd5aed)

This AppWindow.Changed event is subscribed in the TryCompleteInitialization
![Image](https://github.com/user-attachments/assets/e702fc02-c70b-4ded-b38d-ed789c9d8eeb)

But it's only released in the OnUnloaded event
![Image](https://github.com/user-attachments/assets/c07366c3-9932-4734-a1a7-581631b360e2)

If "WebView2.Close" is called before the OnUnloaded and the AppWindow changes position then m_coreWebViewController.NotifyParentWindowPositionChanged(); is called but since the WebView2 has been closed this is no longer allowed and will throw an access violation. 

We were able to work around this by moving our call to Close into the unloaded event and the crash didn't occur.

**Comment by NicholasChrzan** [comment]
@karkarl Do you know if this issue is fixed in 1.6.6? The statement:
"Fixed a crash which would occur on the next AppWindow.Changed event after a WebView2 process failure." seems to be the same root cause as this.

**Comment by codendone** [comment]
@NicholasChrzan Yes, the images you show look like the fix you mentioned. @godlytalias ?
